### DISCUSSION TITLE
When will websocket support be added to v5?

### DISCUSSION THREAD

**Discussion by MichaelNZ85** [adonisjs/core#4047] [Help]
I noticed that v4 supports websockets, but that there's nothing in the official documentation of v5 for this. When is official websocket support coming to v5?

I really really like Adonis - as a Laravel developer it feels so familiar to me, but in TypeScript. The only thing stopping me from using it in my projects is the lack of websocket support - until then I'll stick with Laravel.

**Comment by RomainLanz** [comment]
Hey @MichaelNZ85! 👋🏻 

Unfortunately, there will not be an official package for WebSockets in AdonisJS 5. However, if you require a two-way socket for your application, you can still use third-party libraries such as `Socket.io` or `uWebSockets` within your codebase.

If your goal is simply to send events from your server to your client, it is important to note that WebSockets may not be the best technology for your use case. In this situation, Server-Sent Events (SSE) may be a more suitable solution. You can learn more about SSE here: https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events

It's also worth mentioning that Laravel does not currently support WebSockets. Therefore, if you are comparing AdonisJS to Laravel in terms of WebSocket support, it's important to keep in mind that Laravel does not offer this feature.

**Comment by McSneaky** [comment]
Had small talk about it with Virk yesterday and I think it's not a secret to share

WebSockets was a thing that bought me over from Laravel to NodeJS and Adonis in first place

That WebSocket won't come to V5, but it will make it's way to V6 - which is currently in works. There's no time estimation for it now - it will be released when it's ready, I think it's going to be a thing sometime this year (maybe even first half of the year)

V6 brings a lot of internal cleanup and makes creating and maintaining packages a lot easier :)
